doc/man7/EVP_SIGNATURE-ML-DSA.pod

```diff
@@ -97,26 +108,60 @@ To sign a message using an ML-DSA EVP_PKEY structure:
 
     void do_sign(EVP_PKEY *key, unsigned char *msg, size_t msg_len)
     {
-        size_t sig_len;
+        size_t sig_len = 0;
         unsigned char *sig = NULL;
-        const OSSL_PARAM params[] = {
-            OSSL_PARAM_octet_string("context-string", (unsigned char *)"A context string", 16),
-            OSSL_PARAM_END
-        };
+        OSSL_PARAM params[2];
         EVP_PKEY_CTX *sctx = EVP_PKEY_CTX_new_from_pkey(NULL, pkey, NULL);
         EVP_SIGNATURE *sig_alg = EVP_SIGNATURE_fetch(NULL, "ML-DSA-65", NULL);
 
+        /* The context string is an optional parameter */
+        params[0] = OSSL_PARAM_construct_octet_string("context-string", (unsigned char *)"A context string", 16),
+        params[1] = OSSL_PARAM_construct_end();
         EVP_PKEY_sign_message_init(sctx, sig_alg, params);
         /* Calculate the required size for the signature by passing a NULL buffer. */
         EVP_PKEY_sign(sctx, NULL, &sig_len, msg, msg_len);
-        sig = OPENSSL_zalloc(sig_len);
+        sig = OPENSSL_malloc(sig_len);
         EVP_PKEY_sign(sctx, sig, &sig_len, msg, msg_len);
         ...
         OPENSSL_free(sig);
         EVP_SIGNATURE_free(sig_alg);
         EVP_PKEY_CTX_free(sctx);
     }
 
+To calculate external mu instead of a signature using an ML-DSA EVP_PKEY structure:
+
+    void do_external_mu(EVP_PKEY *key, unsigned char *msg, size_t msg_len)
+    {
+        int mu_only = 1;
+        unsigned char *mu = NULL;
+        size_t mu_len = 0;
+        OSSL_PARAM params[3];
+        EVP_PKEY_CTX *sctx = EVP_PKEY_CTX_new_from_pkey(NULL, pkey, NULL);
+        EVP_SIGNATURE *sig_alg = EVP_SIGNATURE_fetch(NULL, "ML-DSA-44", NULL);
+
+        /* This parameter is required to generate external mu */
+        params[0] = OSSL_PARAM_construct_int(OSSL_SIGNATURE_PARAM_MU_ONLY, &mu_only);
+        /* The context string is an optional parameter */
+        params[1] = OSSL_PARAM_construct_octet_string(OSSL_SIGNATURE_PARAM_CONTEXT_STRING, (unsigned char *)"A context string", 16),
+        params[2] = OSSL_PARAM_construct_end();
+        EVP_PKEY_sign_message_init(sctx, sig_alg, params);
+        /* Calculate the required size for external mu by passing a NULL buffer. */
+        EVP_PKEY_sign(sctx, NULL, &mu_len, msg, msg_len);
+        sig = OPENSSL_malloc(sig_len);
+        /*
+         * As an alternative to using the one-shot EVP_PKEY_sign()
+         * To stream the message you may call
+         *   EVP_PKEY_sign_message_update(sctx, msg_chunk, msg_chunk_len)
+         * multiple times ending with
+         *   EVP_PKEY_sign_message_final(sctx, mu, &mu_len)
+         */
+        EVP_PKEY_sign(sctx, mu, &mu_len, msg, msg_len);
+        ...
+        OPENSSL_free(mu);
+        EVP_SIGNATURE_free(sig_alg);
+        EVP_PKEY_CTX_free(sctx);
+    }
+
```

Comment:
Maybe also an example of computing μ as a signature, and then verifying a signature?